ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,135 @@
+package ch.epfl.ts.evaluation
+
+import scala.language.postfixOps
+import scala.collection.mutable.{Map => MMap, MutableList => MList}
+import scala.concurrent.duration.{DurationInt, DurationLong}
+import scala.concurrent.duration.FiniteDuration
+import akka.actor.{ActorRef, Cancellable}
+import ch.epfl.ts.component.{ComponentRegistration, Component, ComponentRef, StartSignal, StopSignal}
+import ch.epfl.ts.data._
+import ch.epfl.ts.data.Currency._
+
+/**
+  * Evaluates the performance of traders by total returns, volatility, draw down and sharp
+  *
+  * To use this class, redirect all previous connections into trader to instances
+  * of this class.
+  *
+  * @param trader the reference to the real trader component
+  * @param traderId the id of the trader
+  * @param initial the initial seed money
+  * @param currency currency of the intial seed money
+  * @param period the time period to compute performance
+  */
+class Evaluator(trader: ComponentRef, traderId: Long, initial: Double, currency: Currency, period: FiniteDuration) extends Component {
+  // for usage of scheduler
+  import context._
+
+  private var schedule: Cancellable = null
+  private val wallet = MMap[Currency, Double](currency -> initial)
```

Comment:
Maintaining this private copy of the trader's wallet is a duplication of information, because a `Wallet` actor is already being maintained by the `Broker` (with credits & debits on successful transactions, etc)